mdata/aggmetric.go

```diff
@@ -332,18 +252,12 @@ func (a *AggMetric) addAggregators(ts uint32, val float64) {
 
 // write a chunk to persistent storage. This should only be called while holding a.Lock()
 func (a *AggMetric) persist(pos int) {
-	if !cluster.Manager.IsPrimary() {
-		if LogLevel < 2 {
-			log.Debug("AM persist(): node is not primary, not saving chunk.")
-		}
-		return
-	}
-
-	pre := time.Now()
 	chunk := a.Chunks[pos]
+	pre := time.Now()
 
-	if chunk.Saved || chunk.Saving {
-		// this can happen if chunk was persisted by GC (stale) and then new data triggered another persist call
```

Comment:
> Before persist is called, the chunk is closed, which will prevent any future presist() calls from being made.

that prevention is only in GC() though, not in Add(). So the following scenario can still happen as far as i can tell. e.g. : 

* node is primary
* metric hasn't been written to in a while, chunk is open -> GC closes it and calls persist.
* new data comes in for a new chunk -> Add won't close again since already closed, but still calls persist